### PR TITLE
enable snapping_include_static_closure by default for reroute

### LINKED ISSUE #6119: Enable snapping to static closures for the origin of the re-route by default
Tracks enabling `snapping_include_static_closures` for origin of a reroute request by default. Similar to https://github.com/mapbox/mapbox-navigation-android/pull/6050, refs https://github.com/mapbox/mapbox-java/pull/1469#issuecomment-1199860069.

cc @dzinad

### PR DESCRIPTION
**Pull request by dzinad**
### Description
Fixes #6119.

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt [hunk 1]
```diff
@@ -261,6 +244,24 @@ class RouteOptionsUpdater {
         return updatedStartWaypointIndicesIndex
     }
 
+    private fun List<Boolean>?.withFirstTrue(
+        remainingWaypoints: Int,
+        index: Int,
+        coordinatesListSize: Int
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt:250]: NIT

```suggestion
        coordinatesListSize: Int,
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt [hunk 2]
```diff
@@ -100,32 +100,15 @@ class RouteOptionsUpdater {
                                 it.addAll(approachesList.subList(index, coordinatesList.size))
                             }
                         }
-                    ).apply {
-                        if (routeOptions.profile() == DirectionsCriteria.PROFILE_DRIVING_TRAFFIC) {
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt:104]: Was this removed on purpose? Look at https://github.com/mapbox/mapbox-navigation-android/pull/6147

cc @VysotskiVadim

**Review by Guardiola31337 — COMMENTED** [review]
Tests are ❌ 

```
com.******.navigation.core.routeoptions.RouteOptionsUpdaterTest$SnappingClosuresOptionsParameterized > snappingClosuresOptions[4] FAILED
    java.lang.AssertionError at RouteOptionsUpdaterTest.kt:554

com.******.navigation.core.routeoptions.RouteOptionsUpdaterTest$SnappingStaticClosuresOptionsParameterized > snappingClosuresOptions[4] FAILED
    java.lang.AssertionError at RouteOptionsUpdaterTest.kt:675

910 tests completed, 2 failed, 2 skipped

> Task :libnavigation-core:testDebugUnitTest FAILED
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt [hunk 3]
```diff
@@ -100,32 +100,15 @@ class RouteOptionsUpdater {
...
-                                    }
-                                }
-                            )
-                        }
-                    }
+                    )
+                    .snappingIncludeClosuresList(
+                        routeOptions.snappingIncludeClosuresList()
+                            .withFirstTrue(remainingWaypoints, index, coordinatesList.size)
+                    )
```
Comment (LukasPaczos) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt:107]: We cannot rely on nullability when adding `snappingIncludeClosuresList` and `snappingIncludeStaticClosuresList`. We do want to add these parameters to a reroute request even if they were not part of the original request. That's why we need to check for the profile.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt (hunk 2 above)
Comment (dzinad) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt:104]: Rebase issues. Moved it back.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt (hunk 3 above)
Comment (dzinad) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt]: I returned the profile check so it should work as expected now.

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,7 @@ Mapbox welcomes participation and contributions from everyone.
 #### Bug fixes and improvements
 - Updated `NavigationView` to allow drawing of the info panel behind the translucent navigation bar. [#6145](https://github.com/mapbox/mapbox-navigation-android/pull/6145)
 - Optimized vanishing line updates (`MapboxRouteLineApi#updateTraveledRouteLine`) when going through restrictions and legs aren't styled independently. [#6169](https://github.com/mapbox/mapbox-navigation-android/pull/6169)
+- Changed `RouteOptionsUpdater` to use `snapping_include_static_closures=true` for origin of each re-route request. Similar to `snapping_include_closures=true`, but it includes statically closed roads, that is long-term. [#6164](https://github.com/mapbox/mapbox-navigation-android/pull/6164)
```
Comment (VysotskiVadim) [inline CHANGELOG.md:10]:
```suggestion
- The SDK adds `snapping_include_static_closures=true` for an origin of each re-route request. [#6164](https://github.com/mapbox/mapbox-navigation-android/pull/6164)
```
I'm not sure if our users know what `RouteOptionsUpdater` is 🤔

**Review by VysotskiVadim — APPROVED** [review]

**Pull request merged**